ent-orm/usr/usr.go

```diff
@@ -0,0 +1,82 @@
+package usr
+
+import (
+	"context"
+	"fmt"
+
+	"encore.dev/beta/errs"
+)
+
+// CreateUserParams are the parameters to create a user, this cannot use the ent generated types and must use custom
+// struct types for Encore static analysis to work.
+type CreateUserParams struct {
+	Age  int
+	Name string
+}
+
+type Response struct {
+	Message string
+}
+
+// CreateUser is a CRUD endpoint on this service used to create users in the database.
+//
+//encore:api public method=POST path=/users
+func CreateUser(ctx context.Context, params *CreateUserParams) (*Response, error) {
+	client, err := Get()
+	if err != nil {
+		return nil, &errs.Error{
+			Code:    errs.Internal,
+			Message: "Database connection is closed",
+		}
+	}
+
+	created, err := client.User.
+		Create().
+		SetAge(params.Age).
+		SetName(params.Name).
+		Save(ctx)
+	if err != nil {
+		return nil, &errs.Error{
+			Code:    errs.InvalidArgument,
+			Message: "Could not create user",
+		}
+	}
+
+	return &Response{Message: fmt.Sprintf("Fetch this user with %d", created.ID)}, nil
```

Comment:
This would be better. if we responded with `GetUserResponse`

```suggestion
	return &GetUserResponse{
		ID:   created.ID,
		Name: created.Name,
		Age:  created.Age,
	}, nil
```

Reply:
Good idea, the tests are a lot shorter too as a result